pkg/cvo/cvo.go

```diff
@@ -516,18 +532,19 @@ func (optr *Operator) sync(ctx context.Context, key string) error {
 		klog.V(2).Infof("Finished syncing cluster version %q (%v)", key, time.Since(startTime))
 	}()
 
-	// ensure the cluster version exists, that the object is valid, and that
-	// all initial conditions are set.
-	original, changed, err := optr.getOrCreateClusterVersion(ctx, optr.enableDefaultClusterVersion)
+	// ensure the cluster version exists and has not changed
+	original, changed, err := optr.getClusterVersion(ctx)
 	if err != nil {
 		return err
 	}
 	if changed {
 		klog.V(2).Infof("Cluster version changed, waiting for newer event")
 		return nil
 	}
+
+	// this should never happen
 	if original == nil {
-		klog.V(2).Infof("No ClusterVersion object and defaulting not enabled, waiting for one")
+		klog.V(2).Infof("No ClusterVersion object, waiting for one")
```

Comment:
I'm fine dropping this now-dead code, and just making it part of the `getClusterVersion` contract that `original` must be non-`nil` when `err == nil`.  Good excuse to add `getClusterVersion` godocs explaining the contract ;)

Reply:
re-factored and documented